torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2792,19 +2792,21 @@ def sample_inputs_searchsorted(op_info, device, dtype, requires_grad, **kwargs):
     sizes = ((0,), (M,), (0, 0), (M, M), (0, 0, 0), (M, M, M))
     for size, noncontiguous, out_int32, right in product(sizes, [False, True], [False, True], [False, True]):
         unsorted_tensor = make_arg(size, noncontiguous=noncontiguous)
-        input_tensor = make_arg(size, noncontiguous=noncontiguous)
-        if np.product(size) == 0:
-            boundary_tensor = unsorted_tensor
-            sorter = make_tensor(size, dtype=torch.int64, device=device, noncontiguous=noncontiguous)
-        else:
-            boundary_tensor, sorter = torch.sort(unsorted_tensor)
-        side = "right" if right else "left"
+        input_sizes = ((), (M, M), size) if size == (M,) else (size,)
```

Comment:
Instead of adding this `if`, we should refactor to something like below.

```python
# size - (unsorted tensor size, (input sizes,))
sizes = (((0, ), (0, )),
         ((M, ), ((), (M, M), size)),
         ....
```

and everything below stays the same.

It will be a bit more readable and easy to reason about.

Apologies for the late response and Thank you for the ping!

Reply:
done